pkgs/development/compilers/llvm/common/flang-rt/default.nix

```diff
@@ -0,0 +1,78 @@
+{
+  stdenv,
+  llvm_meta,
+  monorepoSrc,
+  src,
+  runCommand,
+  cmake,
+  libclang,
+  libllvm,
+  mlir,
+  ninja,
+  python3,
+  flang-unwrapped,
+  version,
+}:
+let
+  pname = "flang-rt";
+
+  src' =
+    if monorepoSrc != null then
+      runCommand "${pname}-src-${version}" { } (''
+        mkdir -p "$out"
+        cp -r ${monorepoSrc}/cmake "$out"/cmake
+
+        mkdir -p "$out/llvm"
+        cp -r ${monorepoSrc}/llvm/cmake "$out/llvm/cmake"
+        cp -r ${monorepoSrc}/llvm/utils "$out/llvm/utils"
+        cp -r ${monorepoSrc}/third-party "$out/third-party"
+
+        cp -r ${monorepoSrc}/flang-rt "$out"/flang-rt
+        cp -r ${monorepoSrc}/flang "$out"/flang
+        cp -r ${monorepoSrc}/runtimes "$out"/runtimes
+      '')
+    else
+      src;
+
+in
+
+stdenv.mkDerivation rec {
+  inherit pname version;
+
+  src = src';
+
+  sourceRoot = "${src'.name}/runtimes";
+
+  outputs = [
+    "out"
+    "dev"
+  ];
+
+  nativeBuildInputs = [
+    flang-unwrapped
+    cmake
+    ninja
+    python3
+  ];
+  buildInputs = [
+    libclang
+    libllvm
+    mlir
+  ];
+
+  cmakeFlags = [
+    "-DLLVM_DEFAULT_TARGET_TRIPLE=${stdenv.hostPlatform.config}"
+    "-DCMAKE_Fortran_COMPILER=${flang-unwrapped}/bin/flang"
+    "-DCMAKE_Fortran_COMPILER_WORKS=ON"
+    "-DCMAKE_Fortran_COMPILER_SUPPORTS_F90=ON"
+    "-DCLANG_DIR=${libclang.dev}/lib/cmake/clang"
+    "-DMLIR_DIR=${mlir.dev}/lib/cmake/mlir"
+    "-DLLVM_DIR=${libllvm.dev}/lib/cmake/llvm"
+    "-DLLVM_ENABLE_RUNTIMES=flang-rt"
+  ];
```

Comment:
Use `lib.cmake*` functions

Reply:
Done, should be using these across the board now.